### PR TITLE
fix(date formatting): Show study date as a fallback for an empty series date and show 'No Study Date' as a back up for both

### PR DESCRIPTION
**Pull request by jbocce**
### Context

See https://github.com/OHIF/Viewers/issues/3458

### Changes & Results

Show 'No Study Date' in the study browser side panel when no study date is present in the DICOM.

Likewise, in the viewport action bar, when no series date is present, show the study date and if neither are present, show 'No Study Date'.

### Testing

See https://github.com/OHIF/Viewers/issues/3458 for steps to reproduce the original problem.

As a regression test, ensure those studies with study and series dates defined work as expected.

#### PR

- [x] My Pull Request title is descriptive, accurate and follows the
  semantic-release format and guidelines.

#### Code

- [x] My code has been well-documented (function documentation, inline comments,
  etc.)

#### Public Documentation Updates

- [x] The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [x] OS: Windows 11 
- [x] Node version: 16.14.0
- [x] Browser: Chrome 114.0.5735.134
  

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch

### CODE REVIEW

File: extensions/measurement-tracking/src/panels/PanelStudyBrowserTracking/PanelStudyBrowserTracking.tsx [hunk 1]
```diff
@@ -134,7 +134,7 @@ function PanelStudyBrowserTracking({
       const actuallyMappedStudies = mappedStudies.map(qidoStudy => {
         return {
           studyInstanceUid: qidoStudy.StudyInstanceUID,
-          date: formatDate(qidoStudy.StudyDate),
+          date: formatDate(qidoStudy.StudyDate) || 'No Study Date',
```
Comment (jbocce) [inline extensions/measurement-tracking/src/panels/PanelStudyBrowserTracking/PanelStudyBrowserTracking.tsx:137]: This is what was suggested in the reporting issue. However we can discuss alternatives.

Comment (sedghi) [reply extensions/measurement-tracking/src/panels/PanelStudyBrowserTracking/PanelStudyBrowserTracking.tsx:137]: How about N/A?

Comment (jbocce) [reply extensions/measurement-tracking/src/panels/PanelStudyBrowserTracking/PanelStudyBrowserTracking.tsx:137]: N/A could apply to anything... not just a date. If someone sees N/A, they do not know what it is referring to. So no, I do not think N/A is appropriate. How about just 'No Date'.

Comment (jbocce) [reply extensions/measurement-tracking/src/panels/PanelStudyBrowserTracking/PanelStudyBrowserTracking.tsx:137]: Also N/A stands for not applicable. So I really don't think that is right.

Comment (sedghi) [reply extensions/measurement-tracking/src/panels/PanelStudyBrowserTracking/PanelStudyBrowserTracking.tsx:137]: @wayfarer3130 What should be here if it is empty?

Comment (jbocce) [reply extensions/measurement-tracking/src/panels/PanelStudyBrowserTracking/PanelStudyBrowserTracking.tsx:137]: For reference, notice in the video below, the viewport action bar does not wrap when 'No Study Date' is present.


https://github.com/OHIF/Viewers/assets/109477394/489dc352-c375-4704-b507-1a16a7322581

Comment (wayfarer3130) [reply extensions/measurement-tracking/src/panels/PanelStudyBrowserTracking/PanelStudyBrowserTracking.tsx:137]: Probably it should be an empty string if it is missing, then the UI can decide on the value to show when empty.

Comment (jbocce) [reply extensions/measurement-tracking/src/panels/PanelStudyBrowserTracking/PanelStudyBrowserTracking.tsx:137]: @wayfarer3130, this is the UI. For me, the question is what to show in the UI.

Comment (wayfarer3130) [reply extensions/measurement-tracking/src/panels/PanelStudyBrowserTracking/PanelStudyBrowserTracking.tsx]: What about 'No Study Data' or 'No Date'

Comment (wayfarer3130) [reply extensions/measurement-tracking/src/panels/PanelStudyBrowserTracking/PanelStudyBrowserTracking.tsx]: I'm asking Dan about what to show.

**Comment by jbocce** [comment]
> @jbocce - could you update the branch please?

done

File: extensions/measurement-tracking/src/panels/PanelStudyBrowserTracking/PanelStudyBrowserTracking.tsx (hunk 1 above)
Comment (wayfarer3130) [reply extensions/measurement-tracking/src/panels/PanelStudyBrowserTracking/PanelStudyBrowserTracking.tsx]: Agreed with No Study Date

File: platform/core/src/utils/formatDate.js [hunk 2]
```diff
@@ -8,5 +8,6 @@ import moment from 'moment';
  * @returns {string} Formatted date
  */
 export default (date, format = 'DD-MMM-YYYY') => {
-  return moment(date).format(format);
+  // moment(undefined) returns the current date, so return the empty string instead
+  return date ? moment(date).format(format) : '';
```
[the remark this answers is not in the document]
Comment (wayfarer3130) [reply platform/core/src/utils/formatDate.js:12]: No, empty string allows the receiver to format it correctly.

**Review by wayfarer3130 — APPROVED** [review]
Agreed that the No Study Date is the right labelling, so approving for merge.

**Pull request merged**